### Exercise 2.1: Another ball dropped from a tower

A ball is dropped from a tower of height h with initial velocity zero. Write a program that asks the user to enter the height in meters of the tower and then calculates and prints the time the ball takes until it hits the ground, ignoring air resistance. Use your program to calculate the time for a ball dropped from a 100 m high tower.

***

In [8]:
# Cell 1

height = float(input('What is the height of the building, in meters?'))
fall_time = (height / 4.9) ** (0.5)

print('Ignoring air resistance the ball takes', f"{fall_time:.2f}", 'seconds to reach the ground.')

What is the height of the building, in meters? 100


Ignoring air resistance the ball takes 4.52 seconds to reach the ground.


### Exercise 2.2: Altitude of a satellite

A satellite is to be launched into a circular orbit around the Earth so that it orbits the planet once every T seconds.

**a)** Show that the altitude h above the Earth's surface that the satellite must have is
\begin{gather}
    h = \left( \frac{GMT^2}{4\pi^2} \right)^{1/3} - R
\end{gather}
where $G = 6.67 \times 10^{-11} m^{3}kg^{-1}g^{-2}$ is Newton's gravitational constant, $M = 5.97 \times
10^{24} kg$ is the mass of the Earth, and $R = 6371 km$ is its radius.

***

For an object orbiting the Earth in a circular orbit at a constant speed we can make use of the following equation:
\begin{gather*}
    F_c = \frac{mv^2}{r}
\end{gather*}
Where $r$ is measured from the center of the planet. Note that the only force acting is gravity, as such we use Newton's law of universal gravitation.  
\begin{gather*}
    \frac{GMm}{r^2} = \frac{mv^2}{r}\\
    GM = rv^2 \tag{2.1}
\end{gather*}
Since the object is moving in a circular orbit: 
\begin{gather*}
    v = \omega r \tag{2.2}
\end{gather*}
where $\omega$ is the angular frequency, which is of course related to the frequency ($f$) in the following matter: 
\begin{gather*}
    \omega = 2\pi f
\end{gather*}
Which is in turn related to the period
\begin{gather*}
    \omega = \frac{2\pi}{T}
\end{gather*}
Inserting this in equation ($2.2$) we obtain
\begin{gather*}
    v =  \frac{2r\pi}{T}\tag{2.2}
\end{gather*}
Finally, we insert this into equation ($2.1$) 
\begin{gather*}
    GM = \frac{4r^3\pi^2}{T^2}\\
    r^3 = \frac{GMT^2}{4\pi^2}\\
    r = \left( \frac{GMT^2}{4\pi^2}\right)^{1/3}
\end{gather*}
The altitude of the object shall be the distance between the Earth's surface and its coordinate, which leaves us with:
La altitud del objeto seria la distancia entre la superficie de la tierra y la coordenada del objeto:
\begin{gather*}
    h = \left( \frac{GMT^2}{4\pi^2}\right)^{1/3} - R
\end{gather*}
Where ($R$) is the radius of Earth (assuming it is a perfect sphere). 

**b)** Write a program that asks the user to enter the desired value of $T$ and then calculates and prints out the correct altitude in meters.

***

In [15]:
# Cell 2

from math import pi
import numpy as np 

    # Constantes 
G = 6.67 * 10 ** (-11) # Newtons gravitational constant
M = 5.97 * 10 ** (24) # mass of the earth 
R = 6371 * 10 ** 3 # radius of the earth

    # user input for period 
n = int(input('Will you provide a period in, seconds (0), minutes (1) or hours (2)? mark one of the numbers.'))
period = float(input('What is the satellites period?'))

    # for changing to seconds 
if n == 1: # for periods given in minutes 
    period = period * 60 
if n == 2: # for periods given in hours 
    period = period * 60 * 60 

    # Calculation of altitude 
altitude = ((G * M * period ** 2) / (4 * pi ** 2)) ** (1/3) - R

    # answer 
print('The altitude will be ',f"{altitude:,.2f}", 'meters.')

Will you provide a period in, seconds (0), minutes (1) or hours (2)? mark one of the numbers. 2
What is the satellites period? 23.93


The altitude will be  35,773,762.33 meters.


**c)** Use your program to calculate the altitudes of satellites that orbit the Earth once a day (so-called "geosynchronous" orbit), once every 90 minutes, and once every 45 minutes.
What do you conclude from the last of these calculations?

***

We have that a period of $90$ min corresponds to a satellite with an altitude of approximately $279$ km, while a period of $35$ min corresponds to a satellite with an altitude of approximately $-2,181$ km.
The negative sign in the second result implies that our second satellite orbits inside the Earth, which is not possible. As such we can say that in order for a satellite to maintain a circular orbit around the earth, it's period must not be short.

**d)** Technically a geosynchronous satellite is one that orbits the Earth once per sidereal day, which is 23.93 hours, not 24 hours. Why is this? And how much difference will it make to the altitude of the satellite?

***

A period of $24$ hr corresponds to an altitude of $35,855,910.18$ m, while a period of $23.93$ hr corresponds to an altitude of $35,773,762.33$ m. The slight difference in the amount of time it takes for the earth to complete a rotation should come as no shock given the existence of leap years. Between these two altitudes there is a difference of $82,147.85$ m or $0.22\%$